### ISSUE TITLE
structured output to tcp socket

### ISSUE THREAD

**Issue by windelbouwman** [strace/strace#166]
Hi!

This tracing tool is cool! Is there an option to route the events as timestamp/message pairs in either json / cbor format into a tcp/ip pipe? I would like this to be able to visualize them in a trace visualizer.

For example, my own visualization toolie (https://github.com/windelbouwman/lognplot) can ingest (timestamp,message) pairs over a tcp link. For this to work, I would write a python script that takes the strace output, and forwards it in the proper format.

**Comment by windelbouwman** [comment]
Follow up question: what would it take to add a different output format to strace? I looked into the code, and a lot of code boils down to `tprintf`-ing to the currently active `tcb`. Would it be worth it to support different output formats? For example, JSON, XML and these kind of thing? Maybe in a row oriented fashion, such that on each logline an event is emitted?

**Comment by windelbouwman** [comment]
Yup, this would work. A machine parseble structured output in any shape would be great, together with `nc` this would work.

**Comment by rchateauneu** [comment]
>For example, JSON, XML and these kind of thing?

Outputting RDF triples would be great, because strace logs could be fed into a graph database, for example to investigate with SPARQL.
